### ISSUE TITLE
[RFC] Supply config from CLI arguments

### ISSUE THREAD

**Issue by 3tilley** [rochacbruno/dynaconf#538]
Hello!

I don't know if you find the comparison flattering, but I often compare Dynaconf to the way Spring Boot handles config. Spring allows config to be supplied on the [command line](https://docs.spring.io/spring-boot/docs/2.1.9.RELEASE/reference/html/boot-features-external-config.html#boot-features-external-config-command-line-args), and those settings are higher priority than env vars. This is really useful for testing and quick iteration, as well as allowing someone to highlight the most important pieces of config without "hiding" them in config or env vars (like what port a server listens on for example).

Would you consider this, or are there philosophical reasons why not? The only thing I can think is that you then have to get into the business of interacting with whatever CLI tool your users were using, argparse different to click etc. But perhaps you could just supply your own default generated CLI? You could have `--help` display all the configuration options for that app that haven't been specified in config already. This is different to running the dynaconf CLI, and a user of the app wouldn't necessarily even have to know that dynaconf was powering config, just what arguments they needed to pass.

**Comment by rochacbruno** [comment]
One way to achieve this is:

```py
import sys
from dynaconf import Dynaconf
from dynaconf.cli import split_vars

settings = Dynaconf(....)

# load the sys args
settings.update(aplit_vars(sys.argv[1:]))
```

I think we can make it better, so we can implement like this

```py
settings.load_from_cli(**options)
```

I will make some experiments next week.

**Comment by 3tilley** [comment]
Yes I think that would be really good! If you were able to expose a little CLI tool (e.g. click) as well that would make it a lot easier, if nothing else to demonstrate how everything would work with testing
